File: docs/07-archive/legacy-repos/ebs_reverse/docs/01-plan/mockups/capture-diagrams.js
// Playwright script to capture individual diagram sections as PNG
// Usage: npx playwright test capture-diagrams.js (or node with playwright)
const { chromium } = require('playwright');
const path = require('path');

const DIAGRAM_NAMES = [
  'diagram-01-phase-roadmap',
  'diagram-02-system-concept',
  'diagram-03-app-ecosystem',
  'diagram-04-broadcast-prep',
  'diagram-05-hand-sequence',
  'diagram-06-broadcast-end',
  'diagram-07-error-recovery',
  'diagram-08-screen-map',
  'diagram-09-game-distribution',
  'diagram-10-antenna-layout',
  'diagram-11-card-registration',
  'diagram-12-security-mode',
  'diagram-13-priority-grid',
  'diagram-14-roadmap-dependency',
  'diagram-15-viewer-journey',
];

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage({
    viewport: { width: 1000, height: 800 },
  });

  const htmlPath = path.resolve(__dirname, 'clone-prd-diagrams.html');
  const outputDir = path.resolve(__dirname, '..', 'images');

  await page.goto(`file://${htmlPath}`);
  await page.waitForLoadState('networkidle');

  const sections = await page.locator('.diagram-section').all();
  console.log(`Found ${sections.length} diagram sections`);

  for (let i = 0; i < sections.length; i++) {
    const name = DIAGRAM_NAMES[i] || `diagram-${String(i + 1).padStart(2, '0')}`;
    const outputPath = path.join(outputDir, `${name}.png`);

    await sections[i].screenshot({
      path: outputPath,
      scale: 'device',
    });
    console.log(`Captured: ${name}.png`);
  }

  await browser.close();
  console.log(`\nDone! ${sections.length} diagrams captured to ${outputDir}`);
})();

File: docs/07-archive/legacy-repos/ebs_reverse/docs/01-plan/mockups/capture-diagrams.py
"""Playwright script to capture individual diagram sections as PNG."""
import sys
from pathlib import Path
from playwright.sync_api import sync_playwright

DIAGRAM_NAMES = [
    "diagram-01-phase-roadmap",
    "diagram-02-system-concept",
    "diagram-03-app-ecosystem",
    "diagram-04-broadcast-prep",
    "diagram-05-hand-sequence",
    "diagram-06-broadcast-end",
    "diagram-07-error-recovery",
    "diagram-08-screen-map",
    "diagram-09-game-distribution",
    "diagram-10-antenna-layout",
    "diagram-11-card-registration",
    "diagram-12-security-mode",
    "diagram-13-priority-grid",
    "diagram-14-roadmap-dependency",
    "diagram-15-viewer-journey",
]

def main():
    script_dir = Path(__file__).parent
    html_path = script_dir / "clone-prd-diagrams.html"
    output_dir = script_dir.parent / "images"
    output_dir.mkdir(exist_ok=True)

    with sync_playwright() as p:
        browser = p.chromium.launch()
        page = browser.new_page(viewport={"width": 1000, "height": 800})
        page.goto(f"file:///{html_path.as_posix()}")
        page.wait_for_load_state("networkidle")

        sections = page.locator(".diagram-section").all()
        print(f"Found {len(sections)} diagram sections")

        for i, section in enumerate(sections):
            name = DIAGRAM_NAMES[i] if i < len(DIAGRAM_NAMES) else f"diagram-{i+1:02d}"
            out_path = output_dir / f"{name}.png"
            section.screenshot(path=str(out_path), scale="device")
            print(f"  Captured: {name}.png")

        browser.close()
        print(f"\nDone! {len(sections)} diagrams -> {output_dir}")

if __name__ == "__main__":
    main()
